saga/saga.go

```diff
@@ -1,146 +1,200 @@
 package saga
 
-/*
- * Saga Object which provides all Saga Functionality
- * Implementations of SagaLog should provide a factory method
- * which returns a saga based on its implementation.
- */
-type Saga struct {
-	log SagaLog
-}
+import (
+	"sync"
+)
 
-/*
- * Make a Saga which uses the specied SagaLog interface for durable storage
- */
-func MakeSaga(log SagaLog) Saga {
-	return Saga{
-		log: log,
-	}
+// Concurrent Object Representing a Saga
+// Methods update the state of the saga or
+// Provide access to the Current State
+type Saga struct {
+	id       string
+	log      SagaLog
+	state    *SagaState
+	updateCh chan sagaUpdate
+	mutex    *sync.RWMutex
```

Comment:
nope updated, I think I was following an example from somewhere, but can't remember.  Go Sync doesn't specify how to use it so change to non pointer.  

The Go Programming book shows it as not a pointer so looks like that's more idiomatic.  Changed